pkg/server/errors/errors.go

```diff
@@ -35,23 +35,18 @@ type InternalError struct {
 }
 
 func (e InternalError) Error() string {
+	// hide the internal error in the message
 	return e.public.Error()
 }
 
-func (e InternalError) Is(target error) bool {
-	return target.Error() == e.Error()
-}
-
-func (e InternalError) InternalError() string {
-	return e.internal.Error()
-}
-
-func (e InternalError) Internal() error {
+// Unwrap is called by errors.Is. It returns the underlying issue.
+func (e InternalError) Unwrap() error {
 	return e.internal
 }
 
 func (e InternalError) GRPCStatus() *status.Status {
-	return status.New(codes.Code(openfgav1.InternalErrorCode_internal_error), e.public.Error())
+	st, _ := status.FromError(e.public)
```

Comment:
Do we need to do something sane if FromStatus returns not ok?

Reply:
I don't know. I'm open to ideas. As is it today, `internalError.public` will always contain ` a `status.Error` object (line 60), so it's not possible to get a `!ok`.

Reply:
probably if !ok hard code some error?  Just a bit scary we typecast/assert without handling !ok